adobe/react-spectrum
# fix(#4637): Date picker unusable with screen reader (NVDA)

The issue is that NVDA shifts out forms mode when focus lands on the button within a cell within the calendar table, and as such, after announcing the focused element, starts to read the content of the dialog from that point forward. I think the easiest solution for this would be to use `role="application"` instead of `group` for the container `calendarProps` returned by the `useCalendarBase` hook here:  https://github.com/adobe/react-spectrum/blob/7f63e933e61f20891b4cf3f447ab817f918cb263/packages/%40react-aria/calendar/src/useCalendarBase.ts#L94-L97.


Closes #4637 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue 4637](https://github.com/adobe/react-spectrum/issue/4637).

## 📝 Test Instructions:
1. Open https://reactspectrum.blob.core.windows.net/reactspectrum/63ddc5ee08c4e077827ae37bb2323c3c8651850b/docs/react-spectrum/DatePicker.html#example on a Windows machine using NVDA.
2. Navigate to and activate the Calendar dropdown menu button for the DatePicker example labeled "Event date".
3. The Calendar popover dialog will open with focus on the current date. NVDA will be in forms mode and additional dates in the Calendar table after the current should not announce.
4. You should be able to navigate between dates in the Calendar grid using ArrowUp/ArrowDown/ArrowLeft/ArrowRight/Home/End/PageUp/PageDown, and NVDA will stay in forms mode.
5. Closing the Calendar using the Esc key to cancel or Enter to select a date, should restore focus to the Calendar button, and NVDA should exit forms mode.

## 🧢 Your Project:

Adobe/Accessibility
